uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/PublicHoliday.php

```diff
@@ -68,4 +68,39 @@ private static function validateDate($params)
     }
   }
 
+  /**
+   * Returns the number of active Public Holidays between the given
+   * start and end dates (inclusive)
+   *
+   * @param $startDate The start date of the period
+   * @param $endDate The end date of the period
+   * @param bool $excludeWeekends When true it will not count Public Holidays that fall on a weekend. It's false by default
+   *
+   * @return int The Number of Public Holidays for the given Period
+   */
+  public static function getNumberOfPublicHolidaysForPeriod($startDate, $endDate, $excludeWeekends = false)
+  {
+    $startDate = CRM_Utils_Date::processDate($startDate, null, false, 'Ymd');
+    $endDate = CRM_Utils_Date::processDate($endDate, null, false, 'Ymd');
+
+    $tableName = self::getTableName();
+    $query = "
+      SELECT COUNT(*) as public_holidays
+      FROM {$tableName}
+      WHERE date >= %1 AND date <= %2 AND is_active = 1
+    ";
+
+    if($excludeWeekends) {
+      $query .= ' AND DAYOFWEEK(date) BETWEEN 2 AND 6';
```

Comment:
same here 

> also please in case my comment above is not what its really needed by specification and the code will not change ... then replace these magic numbers ( 2 , 6 ) to constants .. I know what they are but someone else may not ) .. it always a good practice not to leave unexplained numbers in the code .